Software/Python/ir_remote_control/gobox_ir_receiver_libs/di_ir_reader.py

```diff
@@ -0,0 +1,486 @@
+#!/usr/bin/env python 
+from subprocess import Popen, PIPE, STDOUT, call
+import time
+import socket
+import time
+
+call("sudo /etc/init.d/lirc stop", shell=True)
+time.sleep(.5)
+
+buf=[]
+######################################
+# IR signals
+# first pulse   : low of ~9000 us 
+# second pulse  : high of ~4200 us
+# 65 pulses between ~500us or ~1600 us for 1 or 0 in the IR signal
+# goes high again until the next pulse comes 
+# all the time is in microseconds
+sig_thresh_len_before_header= 2000 # wait for a signal of at least this much lenght before looking for a signal 
+
+before_header_flag=0
+
+# The signals have a big threshold to compensate for noise coming in the signal
+
+# Noise is of 3 types
+#   * a signal of less than 50 us, we just neglect these noises
+#   * a very small signal which is not measured but causes the reading to stop and start again. We compensate for this by looking for 2 consecutive pulse or space from LIRC, if we see them, then we just add the next signal 
+
+# looking for the 1st signal
+header_thresh=9000
+header_margin=200   #signal length b/w 8800 and 9200
+header_detected_flag=0
+
+# looking for second signal
+header_1_thresh=4200    
+header_1_margin=400     #b/w 3800 and 4600
+header_1_detected_flag=0
+
+trailer_min_length=2000 # looks for the last signal to be atleast 2000us 
+
+noise_thresh=50 # Noise threshold is 50us 
+noise_flag=0
+
+last_pulse_us=0 # last pulse length
+last_sig_type=0 # 1 pulse, 0 space
+add_next_flag=0
+
+pulse=1
+space=0
+
+debug= 0
+detected_sig_buf=[]
+
+p = Popen('mode2 -d /dev/lirc0', stdout = PIPE, stderr = STDOUT, shell = True)
```

Comment:
if this fails, the program does nothing because this https://github.com/karan259/GoPiGo-1/blob/a2c2690fd47d10bd8e0ccb7e792fa5d74cb82654/Software/Python/ir_remote_control/gobox_ir_receiver_libs/di_ir_reader.py#L366 is always empty. I haven;t seen it fail a lot and it would be extremely hard for people to debug it if they break it or try to run it on their own images. We are only going to support it on our image because supporting it on other images might be a nightmare at least for now. @CleoQc @DexterInd : Do you think that is fine